URBNSwiftyConvenience/Classes/URBNBorders.swift

```diff
@@ -0,0 +1,290 @@
+            }
+
+            objc_setAssociatedObject(self, &kURBNBorderViewKey, borderView, .OBJC_ASSOCIATION_RETAIN_NONATOMIC)
+            
+            return borderView
+        }
+        
+        return nil
+    }
+
+    // MARK: Getters
+    private func leftBorder() -> URBNBorder? {
+        return self.urbn_borderView()?.urbn_LeftBorder()
+    }
+    
+    private func rightBorder() -> URBNBorder? {
+        return self.urbn_borderView()?.urbn_RightBorder()
+    }
+    
+    private func topBorder() -> URBNBorder? {
+        return self.urbn_borderView()?.urbn_TopBorder()
+    }
+    
+    private func bottomBorder() -> URBNBorder? {
+        return self.urbn_borderView()?.urbn_BottomBorder()
+    }
+}
+
+public class URBNBorder: NSObject {
+    public var color = UIColor()
+    public var width = CGFloat()
+    public let insets = UIEdgeInsets()
+}
+
+fileprivate final class URBNBorderView: UIView {
+    fileprivate var bv_leftBorder: URBNBorder?
+    fileprivate var bv_rightBorder: URBNBorder?
+    fileprivate var bv_topBorder: URBNBorder?
+    fileprivate var bv_bottomBorder: URBNBorder?
+    
+    private final let width = "width"
+    private final let color = "color"
+    private final let insets = "insets"
+    
+    convenience init() {
+        self.init()
+    }
+    
+    override convenience init(frame: CGRect) {
+        self.init(frame: frame)
+    }
+    
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    // MARK: KVO
+    private func configuredBorder() -> URBNBorder {
+        let border = URBNBorder()
+        border.addObserver(self, forKeyPath: width, options: .new, context: nil)
+        border.addObserver(self, forKeyPath: color, options: .new, context: nil)
+        border.addObserver(self, forKeyPath: insets, options: .new, context: nil)
+        
+        return border
+    }
+    
+    private func unregisterKVO(forBorder border: URBNBorder) {
+        border.removeObserver(self, forKeyPath: width)
+        border.removeObserver(self, forKeyPath: color)
+        border.removeObserver(self, forKeyPath: insets)
+    }
+    
+    override func observeValue(forKeyPath keyPath: String?, of object: Any?, change: [NSKeyValueChangeKey : Any]?, context: UnsafeMutableRawPointer?) {
+        setNeedsDisplay()
+    }
+    
+    // MARK: Clear
+    private func clearAllBorders() {
+        guard
+            let leftBorder = bv_leftBorder,
+            let rightBorder = bv_rightBorder,
+            let topBorder = bv_topBorder,
+            let bottomBorder = bv_bottomBorder else {
+                return
+        }
+        
+        unregisterKVO(forBorder: leftBorder)
+        unregisterKVO(forBorder: rightBorder)
+        unregisterKVO(forBorder: topBorder)
+        unregisterKVO(forBorder: bottomBorder)
+        
+        bv_leftBorder = nil
+        bv_rightBorder = nil
+        bv_topBorder = nil
+        bv_bottomBorder = nil
```

Comment:
bleh, we really should be clearing the associated object with a nil value using `objc_setAssociatedObjects` _(IIRC)_ or else we’ll be leaking memory.